File: src/main/java/net/yakclient/opengl/gui/ComponentRenderingContext.java
package net.yakclient.opengl.gui;

import net.yakclient.opengl.render.GLRenderingContext;

public class ComponentRenderingContext<T extends GLPropsRenderable> implements GLRenderable {
    private final T component;
    private final GUIProperties properties;

    public ComponentRenderingContext(T component, GUIProperties properties) {
        this.component = component;
        this.properties = properties;
    }

    @Override
//    @SuppressWarnings("ConstantConditions")
    public GLRenderingContext[] glRender() {
       return this.component.glRender(this.properties);
//        if (this.component instanceof ScreenRenderer) this.component.glRender(this.properties);
//        if (this.properties.hasProperty(PropertyFactory.CHILDREN_INDEX)) {
//            for (ComponentRenderingContext<?> child :
//                    this.properties.<List<ComponentRenderingContext<?>>>get(PropertyFactory.CHILDREN_INDEX)) {
//                child.glRender();
//            }
//        }
    }
}

File: api/src/main/kotlin/net/yakclient/graphics/api/gui/ComponentRenderingContext.kt
package net.yakclient.graphics.api.gui

import net.yakclient.graphics.api.render.ComponentRenderer
import net.yakclient.graphics.api.render.Renderer
import net.yakclient.graphics.api.render.RenderingContext
import net.yakclient.graphics.api.render.RenderingType

public class ComponentRenderingContext<in T : NativeGuiComponent>(
    private val component: T,
    private val properties: GuiProperties
) : RenderingContext {
    internal fun applyContext() : List<RenderingContext> = component.renderNatively(properties)

    override fun useRenderer(type: RenderingType): Renderer<ComponentRenderingContext<*>> = ComponentRenderer(this, type)
}